### PR TITLE
[HOLD] Improve soft keyboard padding fix with native solution

### PR DESCRIPTION
**Pull request by Julesssss**
### Details
To resolve [a fire](https://expensify.slack.com/archives/C03UW4UU1C5) and ensure we could deploy mobile while in Santiago, we merged a fix for a [keyboard padding issue](https://github.com/Expensify/App/issues/10539) that depends on a `Platform` conditional. 

This PR reverts the original changes and resolves the issue with an improved native fix:

The idea behind this PR is that the underlying cause of the original issue was that both RN and the Android OS were trying to handle the showing/hiding of the soft keyboard. This solution prevents Android from doing this in every single case, ensuring the keyboard control is delegated to RN and the KeyboardAvoidingView. 

The native Android windowSoftInputMode attribute is [explained here](https://developer.android.com/guide/topics/manifest/activity-element#wsoft). We were using `adjustResize`, which would always attempt to resize the Activities View and to make space for the keyboard, whereas `adjustUnspecified` is the default behavior that uses the context of the underlying view. It's important to note that the Android OS will still reduce the Activities View height to make space for the keyboard when RN descides to show/hide it (this is desired, as the alternative is that the entire view moves up to make room for the keyboard). But now, the OS is not forcing the extra padding. 

### Fixed Issues

$ https://github.com/Expensify/App/issues/10539

### Tests

_Please test Android and iOS native apps and mWeb with a physical mobile device_

I have tested on: 
- Physical Nexus 5, Android 8.1
- Physical Pixel 6a, Android 13
- Emulated Pixel 2, Android 8.1
- Simulated iPhone 13, iOS 16.0
- Physical iPhone 7, 13.3 

**1) Ensure the original issue is fixed**
- Create a chat with a user you have not chatted with before
- Tap the input field
- The keyboard should be displayed correctly
- Close the keyboard
- The keyboard should hide correctly

**2) Ensure the keyboard works in other inputs throughout the app**
- Test the app features (registration/sign in/settings/workspace/new chats/ect) ad ensure the soft keyboard works as expected 
- Applause should run the full regression suite instead

**3) Verify that no errors appear in the JS console**

**Phsyical Device screencapture**
_Click to view -- not sure why it's not embedded_
https://user-images.githubusercontent.com/10736861/193265622-fc8eb721-5db1-450b-a4aa-3287e5c62980.mp4

image2

**Emulator Screenshots**
Screenshot_1661437808 Screenshot_1661437843
Screenshot_1661437855

### CODE REVIEW

**Comment by Julesssss** [comment]
Next step is to compile a build and attach it here for Applause to test.

**Comment by Julesssss** [comment]
Reminder to self that this change might actually fix [this other issue](https://github.com/Expensify/App/issues/9252)

**Comment by tgolen** [comment]
What's the status of this PR? Can we get it merged and deployed?

**Comment by Julesssss** [comment]
This is low priority at the moment. But I will return to this. Next step is to compile release builds and test a few of the attributes manually, but this takes a lot of time.

**Comment by tgolen** [comment]
@Julesssss this is actually turning out to be pretty high priority for me since I'm looking into simplifying/improving our KeyboardAvoidingView behaviors. Is there anything I can do to help? I've not ever compiled release builds before. Are these the SOs?

https://stackoverflow.com/c/expensify/questions/7311
https://stackoverflow.com/c/expensify/questions/8290

What do you mean by "test a few of the attributes manually"?

File: src/pages/home/ReportScreen.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import {withOnyx} from 'react-native-onyx';
 import PropTypes from 'prop-types';
-import {Keyboard, Platform, View} from 'react-native';
+import {Keyboard, View} from 'react-native';
```
Comment (Julesssss) [inline src/pages/home/ReportScreen.js:4]: This change seems useless now... 😕

**Comment by Julesssss** [comment]
CC-ing a few people who have context about the original changes that this native fix reverts: @roryabraham @marcaaron @Luke9389

File: src/components/ScreenWrapper/index.android.js [hunk 2]
```diff
@@ -6,13 +6,12 @@ const ScreenWrapper = props => (
...
         {...props}
+        keyboardAvoidingViewBehavior="height"
     >
         {props.children}
     </BaseScreenWrapper>
 );
 
-defaultProps.keyboardAvoidingViewBehavior = 'height';
-
 ScreenWrapper.propTypes = propTypes;
```
Comment (Julesssss) [inline src/components/ScreenWrapper/index.android.js:15]: I am simple reverting [this PR](https://github.com/Expensify/App/pull/10567/files)

File: src/pages/home/ReportScreen.js [hunk 3]
```diff
@@ -235,10 +235,7 @@ class ReportScreen extends React.Component {
         const addWorkspaceRoomErrors = lodashGet(this.props.report, 'errorFields.addWorkspaceRoom');
         const hideComposer = isArchivedRoom || !_.isEmpty(addWorkspaceRoomErrors);
         return (
-            <ScreenWrapper
-                style={[styles.appContent, styles.flex1, {marginTop: this.state.viewportOffsetTop}]}
-                keyboardAvoidingViewBehavior={Platform.OS === 'android' ? '' : 'padding'}
-            >
+            <ScreenWrapper style={[styles.appContent, styles.flex1, {marginTop: this.state.viewportOffsetTop}]}>
```
Comment (Julesssss) [inline src/pages/home/ReportScreen.js:238]: Same here, this is simply reverting [this PR](https://github.com/Expensify/App/pull/10567/files)

**Review by sketchydroide — DISMISSED** [review]

**Comment by Julesssss** [comment]
If anyone wants to make this easier to test, dm me and I'll send you an already signed and compiled APK file to test Android.

**Comment by tgolen** [comment]
OK, I spun up a production build on iOS and I can't get the software keyboard to come up at all. Something must not quite be working right for iOS.

I pinged @Julesssss about it and he said he would look into this further on Monday.

image
image

**Comment by tgolen** [comment]
I'm having issues running a production build on Android (I'm sure it's not related to this PR)

```

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:installRelease'.
> java.util.concurrent.ExecutionException: com.android.builder.testing.api.DeviceException: com.android.ddmlib.InstallException: INSTALL_FAILED_UPDATE_INCOMPATIBLE: Package com.expensify.chat signatures do not match previously installed version; ignoring!
```

**Comment by marcaaron** [comment]
@tgolen try deleting the app and running `npx react-native run-android --variant release` I think if you already have a debug build you'll see this warning.

**Comment by tgolen** [comment]
That worked! Thank you

**Review by Luke9389 — DISMISSED** [review]

**Comment by Julesssss** [comment]
@tgolen I should have realized this on Friday, but it looks like you tested the release build on a simulator -- in which case it's expected that the keyboard won't display 😅 Given the context of the PR it was a good thing to raise though. 

Anyway, after testing I can confirm the keyboard works as expected on a physical device. Added iPhone 7 to the list of devices I have tested. 

I think this is ready for further testing, or to be merged now.

**Review by sketchydroide — APPROVED** [review]

**Review by tgolen — APPROVED** [review]
I think that's some pretty good testing. I'll go ahead and merge this

**Comment by tgolen** [comment]
Actually, holding off to get a C+ review

**Comment by Julesssss** [comment]
> Actually, holding off to get a C+ review

Requested a C+ review [here](https://github.com/Expensify/App/issues/10539#issuecomment-1265695531). Awaiting the review/confirmation.

**Comment by thesahindia** [comment]
@Julesssss, I was able to repro the blank space issue after applying changes.

Steps:-
1. Search new user
2. Keep the keyboard open
3. Tap on the contact

https://user-images.githubusercontent.com/83179295/193702225-660713c6-d7f9-41d6-9a55-3192bcd709a0.mp4

**Comment by Julesssss** [comment]
> @Julesssss, I was able to repro the blank space issue after applying changes.

Damn it. Thank you for testing 😞 

I recall this initially being inconsistently reproducible, but I'll try to see if I can find the cause. I have a feeling it's related to the type of component.

**Comment by tgolen** [comment]
Is that reproducible on any Android? I haven't had the time yet, but I'd be really curious to see if my changes in https://github.com/Expensify/App/pull/11586 will fix that or not.

**Comment by thesahindia** [comment]
> Is that reproducible on any Android? 

Yeah, it was reproducible for iqoo z5. (haven't tested on other android devices)

> but I'd be really curious to see if my changes in https://github.com/Expensify/App/pull/11586 will fix that or not.

I will try testing it again with your changes.

**Comment by Julesssss** [comment]
> I'd be really curious to see if my changes in https://github.com/Expensify/App/pull/11586 will fix that or not.

Oh good point. I'll also test this PR against your changes

**Comment by tgolen** [comment]
What were the results of the testing? Any luck? What's the next step here for this PR?

**Comment by thesahindia** [comment]
> What were the results of the testing? Any luck? What's the next step here for this PR?

Same results. It's still happening 
And as mentioned [here](https://github.com/Expensify/App/issues/10273#issuecomment-1270159682), we are seeing one new issue at login page.

**Comment by tgolen** [comment]
Ah sorry, I got the comments mixed up. Bummer that my PR doesn't fix the empty space issue :(

**Comment by Julesssss** [comment]
> Ah sorry, I got the comments mixed up. Bummer that my PR doesn't fix the empty space issue :(

Yeah... I feel like we're not too far off the underlying issue here, but I'm just not able to spend any further time on this at the moment.

**Comment by PauloGasparSv** [comment]
I could not reproduce this on iOS but it looks similar to this closed issue https://github.com/Expensify/App/issues/11858 so I'm linking it here!

**Comment by tgolen** [comment]
Last time I spoke with Jules about this, the solution here sounded pretty hopeless. We decided to go full-steam-ahead with this PR instead: https://github.com/Expensify/App/pull/11586

I'm going to close this PR. We can reopen it later if we want to.

**Pull request closed without merging**